integration-tests/src/test/java/org/apache/fineract/integrationtests/datatable/DatatableIntegrationTest.java

```diff
@@ -499,37 +499,40 @@ public void validateInsertNullValues() {
-        assertEquals(1, ((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(0));
-        assertEquals("loan_id", ((Map) ((List) items.get("columnHeaders")).get(1)).get("columnName"));
-        assertEquals(loanID, ((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(1));
-        assertEquals("itsABoolean", ((Map) ((List) items.get("columnHeaders")).get(2)).get("columnName"));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(2));
-        assertEquals("itsADate", ((Map) ((List) items.get("columnHeaders")).get(3)).get("columnName"));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(3));
-        assertEquals("itsADatetime", ((Map) ((List) items.get("columnHeaders")).get(4)).get("columnName"));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(4));
-        assertEquals("itsADecimal", ((Map) ((List) items.get("columnHeaders")).get(5)).get("columnName"));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(5));
-        assertEquals("TST_TST_TST_cd_itsADropdown", ((Map) ((List) items.get("columnHeaders")).get(6)).get("columnName"));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(6));
-        assertEquals("itsANumber", ((Map) ((List) items.get("columnHeaders")).get(7)).get("columnName"));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(7));
-        assertEquals("itsAString", ((Map) ((List) items.get("columnHeaders")).get(8)).get("columnName"));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(8));
-        assertEquals("itsAText", ((Map) ((List) items.get("columnHeaders")).get(9)).get("columnName"));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(9));
-
-        assertEquals(2, ((List) ((Map) ((List) items.get("data")).get(1)).get("row")).get(0));
-        assertEquals(loanID, ((List) ((Map) ((List) items.get("data")).get(1)).get("row")).get(1));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(1)).get("row")).get(2));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(1)).get("row")).get(3));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(1)).get("row")).get(4));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(1)).get("row")).get(5));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(1)).get("row")).get(6));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(1)).get("row")).get(7));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(1)).get("row")).get(8));
-        assertNull(((List) ((Map) ((List) items.get("data")).get(1)).get("row")).get(9));
+        List headers = (List) items.get("columnHeaders");
+        List values0 = (List) ((Map) ((List) items.get("data")).get(0)).get("row");
```

Comment:
Can you please rename this to `values`?

Reply:
no, because its the first table row values, and there are also values from the second table row, see line 525

Reply:
I see. `FirstRowValues`, `secondRowValues` then?